test/test_typing.py

```diff
@@ -0,0 +1,396 @@
+        self.assertTrue(result_one.acknowledged)
+        requests_two: List[Union[InsertOne[Movie], ReplaceOne[Movie], DeleteOne]] = [
+            InsertOne(Movie(name="American Graffiti", year=1973)),
+            ReplaceOne(
+                {},
+                {"name": "American Graffiti", "year": "WRONG_TYPE"},
+            ),
+            DeleteOne({}),
+        ]
+        result_two = await coll.bulk_write(requests_two)
+        self.assertTrue(result_two.acknowledged)
+
+    @asyncio_test
+    async def test_command(self) -> None:
+        result: Dict = await self.cx.admin.command("ping")
+        result.items()
+
+    @asyncio_test
+    async def test_list_collections(self) -> None:
+        cursor = await self.cx.test.list_collections()
+        value = cursor.next()
+        value.items()
+
+    @asyncio_test
+    async def test_list_databases(self) -> None:
+        cursor = await self.cx.list_databases()
+        value = await cursor.next()
+        value.items()
+
+    @asyncio_test
+    async def test_default_document_type(self) -> None:
+        client = self.asyncio_client()
+        self.addCleanup(client.close)
+        coll = client.test.test
+        doc = {"my": "doc"}
+        await coll.insert_one(doc)
+        retrieved = await coll.find_one({"_id": doc["_id"]})
+        assert retrieved is not None
+        retrieved["a"] = 1
+
+    @asyncio_test
+    async def test_aggregate_pipeline(self) -> None:
+        coll3 = self.cx.test.test3
+        await coll3.insert_many(
+            [
+                {"x": 1, "tags": ["dog", "cat"]},
+                {"x": 2, "tags": ["cat"]},
+                {"x": 2, "tags": ["mouse", "cat", "dog"]},
+                {"x": 3, "tags": []},
+            ]
+        )
+
+        class mydict(Dict[str, Any]):
+            pass
+
+        result = coll3.aggregate(
+            [
+                mydict({"$unwind": "$tags"}),
+                {"$group": {"_id": "$tags", "count": {"$sum": 1}}},
+                {"$sort": SON([("count", -1), ("_id", -1)])},
+            ]
+        )
+        self.assertTrue(len([doc async for doc in result]))
+
+    @asyncio_test
+    async def test_with_transaction(self) -> None:
+        async def execute_transaction(session):
+            pass
+
+        async with await self.cx.start_session() as session:
+            return await session.with_transaction(
+                execute_transaction, read_preference=ReadPreference.PRIMARY
+            )
+
+
+class TestDocumentType(AsyncIOTestCase):
+    @only_type_check
+    def test_default(self) -> None:
+        client: AsyncIOMotorClient = AsyncIOMotorClient()
+        coll = client.test.test
+        retrieved = coll.find_one({"_id": "foo"})
+        assert retrieved is not None
+        retrieved["a"] = 1
+
+    @only_type_check
+    def test_explicit_document_type(self) -> None:
+        client: AsyncIOMotorClient[Dict[str, Any]] = AsyncIOMotorClient()
```

Comment:
`AgnosticClient` and the other `Agnostic` classes do not allow generic types, but the framework-specific classes do. Is there a way to generalize the agnostic parent types?

Reply:
I'd say we don't pull our hair out trying to get generic types into Motor.